azure/terraform/README.md

```diff
@@ -31,10 +31,34 @@ This is used to help generate unique resource names for:
 * Azure storage account name
 Note: these resources must be globally unique across all of Azure.
 
+The template also provides the `deploy_airflow` and `deploy_argo` flags. These are booleans that specify if airflow or argo will be deployed in the kubernetes cluster along with metaflow related services. By default `deploy_argo` is set to __true__ and `deploy_airflow` is set to __false__.
+
 Next, apply the `infra` module (creates Azure cloud resources only).
 
     terraform apply -target="module.infra"
 
+## Orchestrating Metaflow jobs on Kubernetes
+
+The recommended way to orchestrate Metaflow workloads on Kubernetes is via [Argo Workflows](https://docs.metaflow.org/going-to-production-with-metaflow/scheduling-metaflow-flows/scheduling-with-argo-workflows). However, Airflow is also supported as an alternative.
+
+### Argo
+Argo Workflows is installed by default on the AKS cluster as part of the `services` submodule. Setting the `deploy_argo` [variable](./variables.tf) will deploy argo in the AKS cluster. Not additional configuration is done in the `infra` module to support `argo`.
+
+### Airflow 
+
+**This is not a production deployment template**
+
+If `deploy_airflow`[variable](./variables.tf) is set to true, then the `infra` module will create one more storage blob-container named `airflow-logs` and provide blob-container R/W permissions to the service principal. We create this extra blob-container because Airflow expects the blob-container where it ships logs on Azure to be named `airflow-logs`. 
+
+The `services` module will deploy Airflow via a [helm chart](https://airflow.apache.org/docs/helm-chart/stable/index.html) into the kubernetes cluster (the one deployed by the `infra` module). The Airflow installation will store all the logs in the `airflow-logs` blob-container. The terraform template deploys Airflow configured with a `LocalExecutor`. Metaflow can work with any Airflow executor. This template deploys the `LocalExecutor` for simplicity.
+
+#### Shipping Metaflow compiled DAGs to Airflow
+Airflow expects python files with Airflow dags present in the [dags_folder](https://airflow.apache.org/docs/apache-airflow/2.2.0/configurations-ref.html#dags-folder). By default this terraform template uses the [defaults](https://airflow.apache.org/docs/helm-chart/stable/parameters-ref.html#airflow) set in the Airflow helm chart which is `{AIRFLOW_HOME}/dags` (`/opt/airflow/dags`). 
+
+The metaflow-tools repository also ships a [dag-copy.py](../../scripts/dag-copy.py) file that can help sync airflow dag files generated by Metaflow to the Airflow scheduler _deployed by this template_. Under the hood [dag-copy.py](../../scripts/dag-copy.py) uses the `kubectl cp` command to copy files from local to the Airflow scheduler's container. Example of how to use the file:
```

Comment:
s/airflow/Airflow/g

Also I wonder if this script is really necessary (or desirable).  If we can just show a one-liner shell command directly in README, I think it would be better as it becomes fully transparent what is going on - to channel @oavdeev: "Being asked to run this random script online makes me nervous".

Note: `kubectl` has very elaborate querying features.

If we still end up going with this script, we should rename it something like `airflow_dag_upload.py`

Reply:
Can change the file name. 

The main reason for keeping this file is to simplify putting dags in the airflow scheduler's dags folder. Unfortunately there is no simple way to expose one `kubectl` command.